testing/tests/DevExpress.exporter/jspdfParts/v3/jspdf_v3.dataGrid.styles.tests.js

```diff
@@ -50,7 +50,75 @@ const JSPdfStylesTests = {
                 });
             });
 
-            QUnit.test('Simple - [f1, f2] - custom color in Header', function(assert) {
+            QUnit.test('Simple - [{f1, f2] - RGB color', function(assert) {
+                const done = assert.async();
+                const doc = createMockPdfDoc();
+
+                const dataGrid = createDataGrid({
+                    columns: [
+                        { dataField: 'f1' },
+                        { dataField: 'f2' },
+                    ],
+                    dataSource: [
+                        { f1: 'f1_1', f2: 'f1_2' },
+                    ],
+                });
+
+                const expectedLog = [
+                    'setFillColor,#808080',
+                    'rect,10,15,90,18.4,F',
+                    'text,F1,10,24.2,',
+                    'setFillColor,#808080',
+                    'rect,100,15,80,18.4,F',
+                    'text,F2,100,24.2,',
+                    'text,f1_1,10,42.6,',
+                    'text,f1_2,100,42.6,'
+                ];
+
+                const _rowOptions = {
+                    headerStyles: { backgroundColor: 'rgb(128,128,128)' },
+                };
+
+                exportDataGrid(doc, dataGrid, { topLeft: { x: 10, y: 15 }, columnWidths: [ 90, 80 ], customizeCell, rowOptions: _rowOptions }).then(() => {
+                    // doc.save();
+                    assert.deepEqual(doc.__log, expectedLog);
+                    done();
+                });
+            });
+
+            QUnit.test('Simple - [{f1, f2] - Custom INVALID color', function(assert) {
+                const done = assert.async();
+                const doc = createMockPdfDoc();
+
+                const dataGrid = createDataGrid({
+                    columns: [
+                        { dataField: 'f1' },
+                        { dataField: 'f2' },
+                    ],
+                    dataSource: [
+                        { f1: 'f1_1', f2: 'f1_2' },
+                    ],
+                });
+
+                const expectedLog = [
+                    'text,F1,10,24.2,',
+                    'text,F2,100,24.2,',
+                    'text,f1_1,10,42.6,',
+                    'text,f1_2,100,42.6,'
+                ];
+
+                const _rowOptions = {
+                    headerStyles: { backgroundColor: 'INVALID' },
```

Comment:
It's not clear why our code should convert 'INVALID' to undefined

Reply:
If the user entered an incorrect value, then we must somehow process this request. As example, do not apply this value.

Reply:
Our code paints nothing because all the painting operations are implemented by jsPDF.
It is not clear what we must process.